pkg/controller/sparkapplication/controller_test.go

```diff
@@ -258,7 +258,7 @@ func TestSyncSparkApplication_SubmissionFailed(t *testing.T) {
 	assert.Equal(t, v1alpha1.FailedSubmissionState, updatedApp.Status.AppState.State)
 	assert.Equal(t, int32(1), updatedApp.Status.SubmissionAttempts)
 	assert.Equal(t, float64(0), fetchCounterValue(ctrl.metrics.sparkAppSubmitCount, map[string]string{}))
-	assert.Equal(t, float64(1), fetchCounterValue(ctrl.metrics.sparkAppFailureCount, map[string]string{}))
```

Comment:
can you also have a test for task failure?

Reply:
we have a few tests in https://github.com/lyft/spark-on-k8s-operator/blob/master/pkg/controller/sparkapplication/controller_test.go#L625